p2p/host/basic/basic_host.go

```diff
@@ -112,6 +112,9 @@ var _ host.Host = (*BasicHost)(nil)
 // HostOpts holds options that can be passed to NewHost in order to
 // customize construction of the *BasicHost.
 type HostOpts struct {
+	// EventBus sets the event bus. Will construct a new event bus if omitted.
```

Comment:
I feel uneasy about this.
Maybr we should add the EventBus method to the Network interface and forward from the host so that wr cant end up with two of them.

Reply:
Not really how this could happen, but we can make this change.

Can we defer this until we consolidated all the other repos as well (#1566)? It's happening in the next release.

Reply:
Another option would be to just make this entire thing private.

Reply:
I'm not a big fan of having both a `Config` and `HostOpts`. I expect us to merge these two (or refactor them) in some other way when we start using Fx for service construction (#1993). I suggest not making any big changes in this PR.